src/cargopacket.h

```diff
@@ -188,7 +229,34 @@ struct CargoPacket : CargoPacketPool::PoolItem<&_cargopacket_pool> {
 	inline uint GetDistance(TileIndex current_tile) const
 	{
 		assert(this->source_xy != INVALID_TILE);
-		return DistanceManhattan(this->source_xy, current_tile);
+		assert(this->in_vehicle);
```

Comment:
Using `in_vehicle` outside of `#ifdef WITH_ASSERT` scares me, even though it is in an `assert`.

Reply:
We do this in other places too. But I will add an extra guard, no harm in doing do :)